unity/Assets/Scripts/AgentManager.cs

```diff
@@ -165,91 +165,48 @@ private enum serverTypes { WSGI, FIFO };
+
+            action.autoSimulation = false;
+            physicsSceneManager.MakeAllObjectsMoveable();
+
+            if (action.massThreshold.HasValue) {
+                if (action.massThreshold.Value > 0.0) {
+                    SetUpMassThreshold(action.massThreshold.Value);
+                } else {
+                    var error = "massThreshold must have nonzero value - invalid value: " + action.massThreshold.Value;
+                    Debug.Log(error);
+                    primaryAgent.actionFinished(false, error);
+                    return;
                 }
-        } else if (action.agentMode.ToLower() == "arm") {
-
-            if (action.agentControllerType == "") {
-                action.agentControllerType = "mid-level";
-                Debug.Log("Defaulting to mid-level.");
             }
 
-            if (action.agentControllerType.ToLower() != "low-level" && action.agentControllerType.ToLower() != "mid-level") {
-                var error = "'arm' mode must use either low-level or mid-level controller.";
-                Debug.Log(error);
-                primaryAgent.actionFinished(success: false, errorMessage: error);
-                return;
-            } else if (action.agentControllerType.ToLower() == "mid-level") {
-                // set up physics controller
-                SetUpArmController(true);
-                // the arm should currently be used only with autoSimulation off
-                // as we manually control Physics during its movement
-                action.autoSimulation = false;
-                physicsSceneManager.MakeAllObjectsMoveable();
-
-                if (action.massThreshold.HasValue) {
-                    if (action.massThreshold.Value > 0.0) {
-                        SetUpMassThreshold(action.massThreshold.Value);
-                    } else {
-                        var error = "massThreshold must have nonzero value - invalid value: " + action.massThreshold.Value;
-                        Debug.Log(error);
-                        primaryAgent.actionFinished(success: false, errorMessage: error);
-                        return;
-                    }
+        } else if (action.agentMode.ToLower() == "arm") {
+            SetUpArmController(true);
+            // the arm should currently be used only with autoSimulation off
+            // as we manually control Physics during its movement
+            action.autoSimulation = false;
+            physicsSceneManager.MakeAllObjectsMoveable();
+
+            if (action.massThreshold.HasValue) {
+                if (action.massThreshold.Value > 0.0) {
+                    SetUpMassThreshold(action.massThreshold.Value);
+                } else {
+                    var error = "massThreshold must have nonzero value - invalid value: " + action.massThreshold.Value;
+                    Debug.Log(error);
+                    primaryAgent.actionFinished(success: false, errorMessage: error);
+                    return;
```

Comment:
It seems like lots of this code is heavily duplicated. Can we do something like:
```c#
string agentMode = action.agentMode.ToLower();
...
} else if (agentMode == "stretch" || agentMode == "arm") {
    if (agentMode == "stretch) {
        SetUpStretchController(action);
    } else if (agentMode == "arm") {
        SetUpArmController(true);
    } else {
        throw new InvalidArgumentsException();
    }
    ...
    PREVIOUSLY DUPLICATED CODE
    ....
}
```
Also, it seems like `action.massThreshold` shouldn't be restricted to use by the arm agents. Can we move it outside of these conditional statements so it just happens after initialization no matter which agent is chosen?